met-web/src/components/engagement/dashboard/comment/CommentViewContext.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { createContext, useState, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { getEngagement } from 'services/engagementService';
+import { Engagement } from 'models/engagement';
+import { SubmissionStatus } from 'constants/engagementStatus';
+import { getErrorMessage } from 'utils';
+import { fetchComments } from 'services/commentService';
+import { Comment } from 'models/comment';
+
+export interface EngagementCommentContextProps {
+    engagement: Engagement | null;
+    comments: Comment[];
+    isEngagementLoading: boolean;
+    isCommentsListLoading: boolean;
+}
+
+export type EngagementParams = {
+    engagementId: string;
+};
+
+export const CommentViewContext = createContext<EngagementCommentContextProps>({
+    engagement: null,
+    isEngagementLoading: true,
+    isCommentsListLoading: true,
+    comments: [],
+});
+
+export const CommentViewProvider = ({ children }: { children: JSX.Element | JSX.Element[] }) => {
+    const { engagementId } = useParams<EngagementParams>();
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+
+    const [engagement, setEngagement] = useState<Engagement | null>(null);
+    const [isEngagementLoading, setEngagementLoading] = useState(true);
+    const [isCommentsListLoading, setIsCommentsListLoading] = useState(true);
+    const [comments, setComments] = useState<Comment[]>([]);
+
+    const validateEngagement = (engagementToValidate: Engagement) => {
+        const isClosed = engagementToValidate?.submission_status === SubmissionStatus.Closed;
+
+        if (!isClosed) {
+            throw new Error('Engagement is not yet closed');
+        }
+    };
+    useEffect(() => {
+        const fetchEngagement = async () => {
+            if (isNaN(Number(engagementId))) {
+                navigate('/');
+                return;
+            }
```

Comment:
if (isNaN(Number(engagementId))) { 
I feel like i have seen this code many times ; is there a better way of handling like moving this code to a better place?I am not sure if there is any

Reply:
if (isNaN(Number(engagementId))) { it is used in several places. But I'm not it's big enough that If we created a common component it will help. Like this tells a dev that it is checking if a number is NaN. Pretty clear right? Is replacing it with a common function called verifyParam make it clearer?